akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -254,8 +254,12 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
    * @param events events to be persisted
    * @param handler handler for each persisted `events`
    */
+  final def persistAll[A](events: immutable.Seq[A])(handler: A ⇒ Unit): Unit =
+    events.foreach(persist(_)(handler)) // TODO the atomic part should be handled by issue #15377
+
+  @deprecated("use persistAll instead", "2.4")
   final def persist[A](events: immutable.Seq[A])(handler: A ⇒ Unit): Unit =
-    events.foreach(persist(_)(handler))
+    persistAll(events)(handler)
```

Comment:
I'm tempted to propose immediate deletion instead of deprecation, as it's existence in deprecated status still causes confusion: `persist(List(2))`. If possible I'd vote for removing right away, so the maximum number of breakages come in one version update - M2